staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -40,7 +40,7 @@ var (
 	keepaliveTime    = 30 * time.Second
 	keepaliveTimeout = 10 * time.Second
 	// dialTimeout is the timeout for failing to establish a connection.
-	dialTimeout = 10 * time.Second
+	dialTimeout = 20 * time.Second
```

Comment:
These are really manifest constants then vars.  We should also cross-link the issue or some meta-data to outline how this magic number was chosen.  

We have O(100)s of magic numbers in the system with little/no history.

Reply:
I considered that as well.

The original number seemed to be arbitrary and the new number was just a double of that so equally as arbitrary. When go1.11 is released and is used to build the binaries, this workaround will be moot and can be reverted.

I did reference the hardware to which this was a problem and that this number was chosen as it alleviated that problem for said hardware. There is an issue linked. What further meta-data would you be interested in seeing?